Difficulty compiling on windows

I'm trying to use sync prompt because of ember-tools' dependency on it.  I'm building on windows 7, with MS Visual Studio 12.0 Ultimate installed.  I've tried the following three commands:
 * `npm install -g ember-tools`
 * `npm install -g sync-prompt`
 * `git clone https://github.com/shovon/sync-prompt.git && npm install`

I'm successfully able to compile and install other modules such as bson and node-sass.  I've included my error output here.

```
..\src\sync_prompt.cc(22): error C2065: 'HANDLE' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(22): error C2146: syntax error : missing ';' before identifier 'hStdin' [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(22): error C2065: 'hStdin' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(22): error C2065: 'STD_INPUT_HANDLE' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(22): error C3861: 'GetStdHandle': identifier not found [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(23): error C2065: 'DWORD' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(23): error C2146: syntax error : missing ';' before identifier 'mode' [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(23): error C2065: 'mode' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(24): error C2065: 'hStdin' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(24): error C2065: 'mode' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(24): error C3861: 'GetConsoleMode': identifier not found[C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(27): error C2065: 'mode' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(27): error C2065: 'ENABLE_ECHO_INPUT' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(29): error C2065: 'mode' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(29): error C2065: 'ENABLE_ECHO_INPUT' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(31): error C2065: 'hStdin' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(31): error C2065: 'mode' : undeclared identifier [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
..\src\sync_prompt.cc(31): error C3861: 'SetConsoleMode': identifier not found [C:\Users\Daniel\Development\sync-prompt\build\sync_prompt.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\12.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\Daniel\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1067:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "node" "C:\\Users\\Daniel\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\Daniel\Development\sync-prompt
gyp ERR! node -v v0.12.0
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok
```

--- a/src/sync_prompt.cc
+++ b/src/sync_prompt.cc
@@ -3,7 +3,9 @@
 #include <iostream>
 #include <string>
 
-#ifndef WIN32
+#ifdef WIN32
+#include <windows.h>
+#else
 #include <unistd.h>
 #endif
 

--- a/src/sync_prompt.cc
+++ b/src/sync_prompt.cc
@@ -7,6 +7,7 @@
 #include <windows.h>
 #else
 #include <unistd.h>
+#include <termios.h>
 #endif
 
 using namespace v8;

